fido-device-onboard.spec

```diff
@@ -35,19 +33,18 @@ BuildRequires:  tpm2-tss-devel
 %{summary}.
 
 %prep
+%autosetup -p1 -n fido-device-onboard-rs-0.5.0
 
 %if 0%{?rhel}
-%autosetup -p1 -a1 -n %{name}-rs-%{version}
-rm -f Cargo.lock
 %if 0%{?rhel} >= 10
 %cargo_prep -v vendor
 %else
-%cargo_prep -V 1
+%cargo_prep -V 0
```

Comment:
Generates the vendored source at rpm build time? In the case of RHEL/c9s/eln/Fedora that is not going to work as they don't have out bound connectivity. For copr builds it depends on the repo config but it should be fine for CI builds in copr.

Reply:
From my tests it looks like the `-V 1` specifies the source tarball containing the vendor files. Now those files are contained in the `Source0` tarball and actually works

Reply:
I don't believe we want it all as single tarball.

Reply:
Could you please elaborate further on why we wouldn't want a single tarball?

Reply:
We don't build rhel/c9s using the Makefile, we need an srpm and the vendored tarfile with the dependencies, which is uploaded to the build system.

Reply:
Because that would mean we ship a vendored tarball in Fedora which we don't want and it's the way it's generally done overall within specs. Also it keeps out source separate from the third party stack.

Reply:
We still need the vendored tarball in fedora for ELN builds don't we? and ELN is built from Fedora `dist-git` right? So we need the vendored tarball in all the distributions we build for, don't we?. What's the difference on shipping it in a separate tarball than alongside the source code? Other projects already do that like all the osbuild projects for example.

Reply:
What do the Fedora rust packaging guidelines say? https://docs.fedoraproject.org/en-US/packaging-guidelines/Rust/

I would personally like them to be separate.

Reply:
It literally says:

> [In general](https://docs.fedoraproject.org/en-US/packaging-guidelines/#bundling), packages SHOULD NOT use bundled crate dependencies, whenever possible.

So we are OK in Fedora as we don't use vendoring at all.

Reply:
So using a separate upstream and vendored sourced allows me to:

- Verify the upstream source hasn't changed from the github releases download to verify checksum etc
- Refresh the vendored source against an upstream release without having to change the upstream release source to deal with CVEs in the vendored source.

Having them separate is useful for a number of reasons. Please just update this to keep them separate.

Reply:
> So using a separate upstream and vendored sourced allows me to:
> 
>* Verify the upstream source hasn't changed from the github releases download to verify checksum etc
>

Sorry but I don't get what's the difference. If you include the `vendor` directory within the release you still can verify the checksum exactly the same way.
 
>* Refresh the vendored source against an upstream release without having to change the upstream release source to deal with CVEs in the vendored source.
> 

Applying a CVE downstream without having to modify the upstream repo would be as simple as:
```console
git clone https://github.com/fdo-rs/fido-device-onboard-rs.git
cd fido-device-onboard-rs
git checkout v0.5.0
# do the required modifications to address the CVE by changing dependencies and/or adding patches to the `patches` directory
./update-vendor.sh
git diff > cve-number.diff 
```
And use that `cve-number.diff` as patch in the dowstream repo.

> 
> Having them separate is useful for a number of reasons. Please just update this to keep them separate.

Sorry but I cannot update the PR to separate them, that's the whole point of this PR: to minimize hacks and to bring the filtered and patched vendor directory into the repo's source code. The only reason I see for now to not merge it is the problem I found regarding `cargo add` that I mentioned below. However that would be a minor problem that can be solved by:
```
rm -f .cargo/config.toml
cargo add ...
./update-vendor.sh
```